services/QuillLMS/app/lib/error_notifier.rb

```diff
@@ -6,4 +6,19 @@ def self.report(error, options = {})
     Sentry.capture_exception(error, extra: options)
     NewRelic::Agent.notice_error(error, options)
   end
+
+  def self.report_long_running(error, threshold, options = {})
+    start = current_time
+    result = yield
```

Comment:
The current code layout does require the return.  The caching takes place outside of the code that's being benchmarked, which means that the result of the benchmarking block needs to bubble up in `return` chains so that the originally calling function can put the results in the cache.  We could refactor this to do the cache writing inside the benchmarking block instead if you think not returning values from benchmarking is worth it.